### PR TITLE
feat(bridge): control-message dispatcher + GUI prompts + auth-required exit 87 (Phase 1, PR 1.9)

### PR DESCRIPTION
**Pull request by AlexDochioiu**
## Summary

Phase 1, PR 1.9 of the desktop plan (`docs/desktop/PLAN.md`): three tightly-coupled pieces of the supervised-mode inbound/prompt seam. All gated by `--control-url`; standalone behaviour is byte-identical.

### 1. `BridgeControlMessageDispatcher` (Layer 4 `control/`)
The **single** inbound subscriber/decoder for GUI→helper control messages. Decodes each frame exactly once (warn+skip on undecodable, preserved from the token service's old decode point) and routes via an exhaustive switch:
- `token_response` / `token_update` → `ControlChannelTokenService` typed delegates
- `prompt_response` → `ControlPromptService` delegate
- everything else ignored — `restart` is helper→GUI only and is **never** an inbound command; `unregister_and_exit` gains its route in PR 1.11.

`ControlChannelTokenService` no longer subscribes to `inbound`; it gained `handleTokenResponse`/`handleTokenUpdate` delegates while keeping its correlation map, issue-sequence ordering, `token_request` send path, timeout and dispose semantics unchanged. The single-subscriber property is asserted by test (listen-counting fake client).

### 2. Prompts become structured control events
New `ControlPromptService` (Layer 3 `services/`, the same blessed `ControlChannelClient` seam as the token service) owns prompt-class correlation and **all** prompt-class sends. It implements the new `BridgeReplacePrompt` interface (`server/foundation/`; `TerminalPromptRepository` implements it too — two production impls), and `BridgeInstanceService` now takes `BridgeReplacePrompt replacePrompt`. A supervised bridge therefore surfaces replace-bridge contention to the desktop GUI (accepted→replace, rejected→decline; channel down / 2-min timeout / teardown → `nonInteractive`, mirroring the terminal's "couldn't ask"). The logout CLI keeps the terminal implementation (never supervised).

### 3. Auth-required exit sentinel `87` (ADR A23)
When the GUI cannot supply a token at bootstrap (`ControlTokenUnavailableException` — signed out / mid-login / unreachable), the supervised bridge logs once, sends a best-effort `loginNeeded` prompt (fire-and-forget, swallow+log when the channel is down) and exits **87** — distinct from restart (86), control-loss (1), crash (other) — so the Phase-2 GUI state machine can prompt for login instead of backoff-respawn-thrashing. The sentinel is wired into the shutdown backstop chain and the outer-finally shutdown-error guard, so a hung or throwing shutdown still reports 87 (same robustness 86 has).

## Regression guide (re-verified)

1. Token pull round-trip, force-refresh, push adoption, null-response cache invalidation, dispose-fails-in-flight: all 19 existing token-service behaviour tests kept their expectations — only re-wired through a dispatcher over the same fake client.
2. `ControlChannelLossListener` untouched (separate `connectionState` subscription).
3. Undecodable-frame warn+skip preserved at the single decode point (dispatcher test).
4. Standalone Console output byte-identical — no Console call sites changed; nothing supervised is constructed without `--control-url`; `TerminalPromptRepository` behaviour untouched.
5. Supervised exit codes: token-unavailable at bootstrap = 87; control-loss = 1 (unchanged); restart = 86 (unchanged); normal shutdown = 0.

## Tests

- New `bridge_control_message_dispatcher_test.dart` (7) + `control_prompt_service_test.dart` (9).
- `make analyze` + `dart analyze --fatal-infos` (bridge/app) clean; `make test` all pass (app: 1627).

## Plan tracking

PLAN.md pointer + §9 row flipped; §6 gained `ControlPromptService` / `BridgeReplacePrompt` rows; phase doc Findings + Deltas recorded (incl. the deliberate scoping of "essential Console output" to prompt-class events only — the DTO surface has no generic console variant; stdout/stderr stays the log path for the PR-2.6 capture).

## Aristotle

- plan-review: APPROVED
- impl-review: APPROVED

---
## Summary by cubic
Introduces a single control-channel dispatcher, moves user prompts to the desktop app, and adds exit code 87 when auth is required during supervised startup. Gated by `--control-url`; standalone behavior is unchanged.

- **New Features**
  - Single inbound `BridgeControlMessageDispatcher` routes `token_response`, `token_update`, and `prompt_response`; warns and skips undecodable frames; ignores non-inbound commands like `restart`.
  - New `ControlPromptService` sends GUI prompts and tracks replies; degrades to `nonInteractive` if the channel is down, times out, or a transport error occurs; implements `BridgeReplacePrompt` so `BridgeInstanceService` asks the GUI in supervised mode; the logout CLI keeps the terminal prompt.
  - Supervised bootstrap exits with sentinel 87 when the GUI cannot supply a token; sends a best-effort `loginNeeded` prompt and preserves the sentinel through shutdown; restart 86 unchanged.

- **Refactors**
  - `ControlChannelTokenService` no longer subscribes to inbound; adds `handleTokenResponse`/`handleTokenUpdate` delegates; request correlation, ordering, timeouts, and dispose behavior unchanged.
  - Dispatcher is started before the bootstrap token pull; prompt copy centralized in `BridgeReplacePrompt` constants; supervised wiring lives behind `--control-url`.
  - Fixed bootstrap token-failure logging to include the caught error.
  - Added tests for the dispatcher and prompt service; updated token-service tests to run through the dispatcher.

<sup>Written for commit aca62491b95793d66999b86c64c0b070b7a83617. Summary will update on new commits.</sup>

<a href="https://cubic.dev/pr/sesori-ai/sesori_apps_monorepo/pull/371?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

### CODE REVIEW

File: bridge/app/lib/src/services/control_prompt_service.dart [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+    required ControlChannelClient client,
+    Duration responseTimeout = _defaultResponseTimeout,
+  })  : _client = client,
+        _responseTimeout = responseTimeout;
+
+  @override
+  Future<TerminalPromptDecision> askReplaceExistingBridge({required int bridgeCount}) {
+    return _ask(
+      kind: ControlPromptKind.replaceBridge,
+      message: "Another Sesori bridge is already running. Kill it and start fresh?",
```
[the remark this answers is not in the document]
Comment (AlexDochioiu) [reply bridge/app/lib/src/services/control_prompt_service.dart]: [Sesori reply] Addressed (in commit 2af95a9d)

Extracted the shared question copy into `BridgeReplacePrompt.replaceExistingBridgeMessage` / `replaceStartingBridgeMessage(holderPid:)` next to the interface; both `TerminalPromptRepository` (which appends its own `[y/N]` hint) and `ControlPromptService` now reference them, so the two paths can't drift.

File: bridge/app/lib/src/services/control_prompt_service.dart [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+      _client.send(
+        jsonEncode(
+          ControlMessage.promptRequest(
+            id: id,
+            kind: ControlPromptKind.loginNeeded,
+            message: "Sign in to the Sesori desktop app to start the bridge.",
+          ).toJson(),
+        ),
+      );
+    } on ControlChannelNotConnectedException catch (error) {
```
[the remark this answers is not in the document]
Comment (AlexDochioiu) [reply bridge/app/lib/src/services/control_prompt_service.dart:75]: [Sesori reply] Addressed (in commit 2af95a9d)

`announceLoginNeeded` now also has an `on Object` catch (logged with error + stack trace) after the specific not-connected case, so any unexpected transport failure is swallowed and the caller still returns the authoritative exit code 87. Covered by a new test using a `StateError`-throwing send.

File: bridge/app/lib/src/services/control_prompt_service.dart [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+      Log.w("[control][prompt] could not announce login-needed — control channel down", error);
+    }
```
[the remark this answers is not in the document]
Comment (AlexDochioiu) [reply bridge/app/lib/src/services/control_prompt_service.dart:82]: [Sesori reply] Addressed (in commit 2af95a9d)

Added an `on Object` catch (logged with error + stack trace) after the specific `ControlChannelNotConnectedException` case, so any unexpected transport failure stays best-effort and never masks the authoritative exit code 87. New test covers a `StateError`-throwing send.

File: bridge/app/lib/src/services/control_prompt_service.dart [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+        _client.send(
+          jsonEncode(ControlMessage.promptRequest(id: id, kind: kind, message: message).toJson()),
+        );
+      } on ControlChannelNotConnectedException catch (error) {
+        // The GUI is unreachable (outage / mid-reconnect): the question cannot
+        // be asked. Degrade like a non-interactive terminal; logged because the
+        // transport failure is otherwise swallowed into a decision.
+        Log.w("[control][prompt] could not send prompt — control channel down", error);
+        return TerminalPromptDecision.nonInteractive;
+      }
```
[the remark this answers is not in the document]
Comment (AlexDochioiu) [reply bridge/app/lib/src/services/control_prompt_service.dart:136]: [Sesori reply] Addressed (in commit 2af95a9d)

The `_ask` send path now also catches `on Object` (logged with error + stack trace) and degrades to `TerminalPromptDecision.nonInteractive`, keeping the specific not-connected catch for its clearer message. New test covers the unexpected-failure path.

File: bridge/app/lib/src/bridge/runtime/bridge_runtime_runner.dart [hunk 5]
```diff
@@ -317,7 +354,18 @@ class BridgeRuntimeRunner {
...
       if (supervisedTokenService != null) {
-        authAccessToken = await supervisedTokenService.getAccessToken();
+        try {
+          authAccessToken = await supervisedTokenService.getAccessToken();
+        } on ControlTokenUnavailableException catch (error) {
+          // The GUI cannot supply a token (signed out / mid-login / down). Exit
+          // with the auth-required sentinel so the GUI prompts for login instead
+          // of backoff-respawning a helper that can never start. The prompt is
+          // advisory (best-effort); the exit code is the authoritative signal.
+          Log.e("Cannot start supervised: $error");
```
[the remark this answers is not in the document]
Comment (AlexDochioiu) [reply bridge/app/lib/src/bridge/runtime/bridge_runtime_runner.dart]: [Sesori reply] Addressed (in commit aca62491)

The auth-required path now logs `Log.e("Cannot start supervised — no access token from the desktop app", error, stackTrace)`, passing the caught exception and stack trace as structured logger arguments instead of interpolating it.
